sql/catalyst/src/main/java/org/apache/spark/sql/vectorized/ArrowColumnVector.java

```diff
@@ -508,4 +516,38 @@ final ColumnarMap getMap(int rowId) {
       super(vector);
     }
   }
+
+  private static class IntervalYearAccessor extends ArrowVectorAccessor {
+
+    private final IntervalYearVector accessor;
+
+    IntervalYearAccessor(IntervalYearVector vector) {
+      super(vector);
+      this.accessor = vector;
+    }
+
+    @Override
+    int getInt(int rowId) {
+      return accessor.get(rowId);
+    }
+  }
+
+  private static class IntervalDayAccessor extends ArrowVectorAccessor {
+
+    private final IntervalDayVector accessor;
+    private final NullableIntervalDayHolder intervalDayHolder = new NullableIntervalDayHolder();
+
+    IntervalDayAccessor(IntervalDayVector vector) {
+      super(vector);
+      this.accessor = vector;
+    }
+
+    @Override
+    long getLong(int rowId) {
+      accessor.get(rowId, intervalDayHolder);
+      return Math.addExact(
+              intervalDayHolder.days * MICROS_PER_DAY,
```

Comment:
Both multiplications can overflow too. Could you use `Math.multiplyExact`, please.

Reply:
OK，Thanks @MaxGekk

Reply:
@MaxGekk done